### ISSUE TITLE
Map Beacons are currently non-functional

### ISSUE THREAD

**Issue by RyanStrudfelt** [space-wizards/space-station-14#27204]
## Description
At present, on all stations Map Beacons do not function. Even though they exist on the station they do not appear on handheld maps or wallmounted station maps. 

**Reproduction**
Use any Station map

**Screenshots**
![image](https://github.com/space-wizards/space-station-14/assets/84740613/c996ac58-dca3-4225-bc6c-b5c0a85031a1)

**Comment by MozarteanChaos** [comment]
Something that might be helpful to know is that renaming the beacon makes it function properly.

**Comment by deltanedas** [comment]
locale mapinit thing if true

**Comment by Errant-4** [comment]
I do not think this is a locale issue, because then the locale-string would be displayed instead ((which is also a thing that happens, but there is a separate issue for that)) . This is what it looks like when a locale string is not resolved (medical)
![image](https://github.com/space-wizards/space-station-14/assets/35878406/596ab7f0-7684-4af2-b783-02ec807054c4)

**Comment by deltanedas** [comment]
the beacon component stores locale id and resolved text differently, it gets resolved on mapinit

**Comment by Errant-4** [comment]
> the beacon component stores locale id and resolved text differently, it gets resolved on mapinit

NavMapBeaconsComponent has a `Text` datafield. It starts out containing the prototype's particular locale id, and when the beacon entity is in the mapinit phase it's supposed to have Text be rewritten with the resolved text of the locale id. So when someone spawns a beacon, by the time they look at it .Text is supposed to just contain the localised label for that beacon

Could be useful to know if Text is actually empty when this bug manifests, or its just not shown on the map for some reason. But I have not been able to repro it

**Comment by deltanedas** [comment]
i think what causes it it is a pro renaming a beacon makes every other beacon not have a name, only the one they renamed

**Comment by Errant-4** [comment]
> i think what causes it it is a pro renaming a beacon makes every other beacon not have a name, only the one they renamed

I have never seen this issue with a single beacon being visible. When they are gone, they are all gone. Renaming also did not break them in Dev

**Comment by Errant-4** [comment]
I spent ~4 hours staring at a map on Live today, and finally managed to get one replication. It appears that the issue may be related to multiple people accessing a map at the same time (although there were multiple instances of that happening before, with no issue)


```

Lizard Round 55131  (times are Shift Time from the PDA)

53:12 
  I am on the starboard end of the corridor just above the bridge, standing next to the map. I have the map window open. Kajin-Bex is also standing there and may or may not have the map open
  Kajin-Bex says: "Ya'll joining us at the screen""
53:13 
  Map window closes by itself
  I am distracted by talking to Kajin-Bex and don't notice
53:44 
  I open the map window again, and the labels are all missing
```

I raised this in Ahelp and Nik started looking at the logs, but I think he hasn't found anything useful yet

Dev testing with 2 clients has revealed some anomalous behaviour, but could not yet cause the beacons to disappear

If 2 players open the same Map UI, one of them can walk away and have the window remain open at any range, so long as the other player still has the window open on their end. Sometimes the map will blink for the remaining player (possibly when the other player walks out of range? When they close the map? I have not been able to repro that in testing, either, but experienced it multiple times on live when other people used "my" wallmap)
